lib/identity_cache/cached/recursive/association.rb

```diff
@@ -47,6 +52,32 @@ def embedded_by_reference?
         def embedded_recursively?
           true
         end
+
+        private
+
+        def fetched?(records)
+          record = records.first
+          super || record.instance_variable_defined?(dehydrated_variable_name)
+        end
+
+        def fetch_embedded(records)
```

Comment:
This isn't just used for recursively embedded associations.  It is also used for id embedded associations to get the associated ids.